deep_qa/layers/adaptive_recurrence.py

```diff
@@ -67,7 +67,7 @@ class AdaptiveStep(Layer):
     probabilities, states and outputs from a timestep t's contribution if they have already reached
     1 - epsilon at a timestep s < t.
     '''
-    def __init__(self, memory_network: 'MemoryNetworkModel', layer_params: Dict[str, Any],
```

Comment:
Forward reference (just marking all of these with a comment so that they can be fixed once we decide on a solution).